invenio_rdm_records/resources/deserializers/rocrate/schema.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RO-Crate schema."""
+
+from marshmallow import EXCLUDE, Schema, fields, pre_load
+from marshmallow_utils.fields import SanitizedHTML, SanitizedUnicode
+
+
+def _list_value(lst):
+    """Return the first value from single-value lists."""
+    if isinstance(lst, list) and len(lst) == 1:
+        return lst[0]
+    return lst
+
+
+class ROCrateSchema(Schema):
+    """Schema for the loading RO-Crate into record metadata."""
+
+    class Meta:
+        """Meta attributes for the schema."""
+
+        unknown = EXCLUDE
+
+    LIST_VALUE_FIELDS = [
+        "license",
+        "description",
+        "datePublished",
+    ]
+
+    @pre_load
+    def list_to_value(self, data, **kwargs):
+        """Extract top-level single-item list field values."""
+        for field in self.LIST_VALUE_FIELDS:
+            if field in data:
+                data[field] = _list_value(data[field])
+        return data
+
+    resource_type = fields.Constant({"id": "dataset"})
+    title = SanitizedUnicode(data_key="name", required=True)
+    description = SanitizedHTML()
+    publication_date = fields.Method(
+        deserialize="load_publication_date",
+        data_key="datePublished",
+    )
+
+    def load_publication_date(self, value):
+        """Load publication date."""
+        if value:
+            return value[:10]
+
+    creators = fields.Method(
+        deserialize="load_creators",
+        required=True,
+        data_key="author",
+    )
+
+    def load_creators(self, value):
+        """Load creators."""
+        if value and isinstance(value, list):
```

Comment:
moderate: `else`? Should throw a `ValidationError`? or process one and append as list?